### DISCUSSION TITLE
Rack Not Found?

### DISCUSSION THREAD

**Discussion by KSBolton** [netbox-community/netbox#6234] [Q&A] [answered]
Hello, I'm currently on v2.11.1 but I believe since 2.11.0 I've been hitting this error while trying to bulk import devices. I've created the CSV with the necessary headers and information but Netbox tells me that my rack cannot be found even though it definitely exists. Am I doing something wrong?

![IMG_20210421_101719_092](https://user-images.githubusercontent.com/42079864/115578333-d0f78100-a28a-11eb-93d8-73b3e5328452.jpg)
![IMG_20210421_101703_848](https://user-images.githubusercontent.com/42079864/115578340-d1901780-a28a-11eb-8c75-2b0cf09a0836.jpg)

**Comment by candlerb** [comment, marked as answer]
I've just tested this. Solution: If RACK-2-APC is within a rack group (now called "location"), then you must specify that at the same time by including a "location" column in the CSV.

Checking the SQL, I see that rack names are unique per location, which explains why locations must be supplied together with rack name:

```
netbox=# \d dcim_rack
...
    "dcim_rack_group_id_name_846f3826_uniq" UNIQUE CONSTRAINT, btree (location_id, name)
```

I don't know if this was a behaviour change in 2.11; I don't have an older version to compare against.

**Reply by candlerb** [reply]
Actually, I do have access to a v2.10.10.  I see the same: if the rack belongs to a RackGroup, but I don't include a "rack_group" column in the CSV, then the rack isn't found.

**Reply by KSBolton** [reply]
You're absolutely right. Thank you again!
